core/templates/dev/head/collection_editor/collection_editor.html

```diff
@@ -80,5 +173,10 @@
     {{ include_js_file('domain/editor/ChangeObjectFactory.js') }}
     {{ include_js_file('domain/editor/UndoRedoService.js') }}
     {{ include_js_file('domain/utilities/UrlInterpolationService.js') }}
+    {{ include_js_file('editor/ExplorationEditor.js') }}
```

Comment:
Yeah, seeing this makes me agree with your thoughts on the issue thread. The collection editor really shouldn't be dependent on ExplorationEditor, EditorServices, GadgetValidationService, or expressionInterpolationService.

At this point, I'm thinking we should write a new EditorRouterService to address this. Looking at the old RouterServices file, it is doing some behaviors which I don't think we want to replicate in the collection editor, anyway. I think for starters, a new EditorRouterService should support registering which tabs (and their relative hrefs) are supported by the editor. Also, the old router service scanned the URL for figuring out which tab is selected. This seems a bit ick to me. Switching tabs should involve changing the href and simply tracking which tab is currently selected. If you go to a collection editor tab directly, then which tab is selected can be passed through the backend controller. I feel like, overall, this makes the routing service less fragile, though I'm very keen to hear your thoughts on it.